### ISSUE TITLE
Allow plugin issues to specify custom documentation URLs

### ISSUE THREAD

**Issue by alies-dev** [vimeo/psalm#11732]
## Summary

Plugin-defined issues (subclasses of `PluginIssue`) have no mechanism to provide custom documentation URLs. The current `link` field in `IssueData` is derived exclusively from `SHORTCODE` and hardcoded to the `psalm.dev` domain, which is not meaningful for third-party plugin issues.

## Current behavior

In `IssueData::__construct()` ([IssueData.php, line ~49](https://github.com/vimeo/psalm/blob/master/src/Psalm/Internal/Analyzer/IssueData.php#L49)):

```php
$this->link = $shortcode ? 'https://psalm.dev/' . str_pad((string) $shortcode, 3, "0", STR_PAD_LEFT) : '';
```

- Built-in Psalm issues use `SHORTCODE` to generate links like `https://psalm.dev/042`, which resolve to documentation pages.
- Plugin issues inherit `SHORTCODE = 0` from `CodeIssue`, resulting in an empty link.
- Even if a plugin overrides `SHORTCODE`, the generated URL would point to `psalm.dev`, which has no documentation for third-party issues.

## Expected behavior

Plugin authors should be able to associate documentation URLs with their custom issues. When a plugin issue is reported (in console output, SARIF, or other report formats), the link should point to the plugin's own documentation explaining the issue and how to resolve it.

## Suggested approach

Introduce an overridable constant or method on `CodeIssue` that allows specifying a custom documentation URL:

```php
abstract class CodeIssue
{
    public const DOCUMENTATION_URL = '';
    // ...
}
```

Then in `IssueData`:

```php
$this->link = match (true) {
    $documentation_url !== '' => $documentation_url,
    $shortcode > 0 => 'https://psalm.dev/' . str_pad((string) $shortcode, 3, "0", STR_PAD_LEFT),
    default => '',
};
```

This would allow plugin authors to write:

```php
final class NoEnvOutsideConfig extends PluginIssue
{
    public const DOCUMENTATION_URL = 'https://github.com/psalm/psalm-plugin-laravel/blob/master/docs/issues/NoEnvOutsideConfig.md';
}
```

Alternatively, a method-based approach (`getDocumentationUrl(): string`) would also work and provides more flexibility for dynamic URL generation.

## Context

Discovered while developing a custom rule for [psalm-plugin-laravel](https://github.com/psalm/psalm-plugin-laravel). The plugin defines issue classes that users need guidance on, but there is no way to link them to the plugin's documentation.

**Comment by danog** [comment]
LGTM!

**Comment by alies-dev** [comment]
ready in 7.x 🚀 🚀 🚀  And laravel plugin already use it: https://github.com/psalm/psalm-plugin-laravel/blob/d0758afa5c78d18694de0225ea4c8e22984a8cbf/src/Issues/InvalidConsoleOptionName.php#L9-L16
